io/src/lib.rs

```diff
@@ -73,7 +78,10 @@ pub trait Read {
     #[doc(alias = "read_to_end")]
     #[cfg(feature = "alloc")]
     #[inline]
-    fn read_to_limit(&mut self, buf: &mut Vec<u8>, limit: u64) -> Result<usize> {
+    fn read_to_limit(&mut self, buf: &mut Vec<u8>, limit: u64) -> Result<usize>
+    where
+        Self: Sized,
```

Comment:
You should be able to call `take()` on `&mut R`, so this is not needed.

Reply:
I had two goes and couldn't work it out. I force pushed without all the `bitcoin` changes.

Reply:
Oh, that's because of the missing impl. We need to have `type ByRef<'a, R> = &'a mut R` in some future release. Anyway, `ByRef` should work for now.